person.rb

```diff
@@ -0,0 +1,30 @@
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'corrector'
+
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..10.000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    @corrector = Corrector.new
+  end
+
+  def can_use_services?
+    is_of_age or @parent_permission
+  end
+
+  def validate_name
+    @corrector.correct_name(@name)
```

Comment:
- [ ] This method should use the corrector instance to validate and save the corrected name like, `@name = @corrector.correct_name(@name)`. Please kindly fix that.

Reply:
Done!